### PR TITLE
add support for serving via ssl

### PR DESCRIPTION
**Pull request by jakedetels**
Allows users to run server using SSL via `broccoli serve -ssl`.

SSL key and certificate files must be located in the user's project root at `ssl/server.key` and `ssl/server.crt`.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ function broccoliCLI () {
     .description('start a broccoli server')
     .option('--port <port>', 'the port to bind to [4200]', 4200)
     .option('--host <host>', 'the host to bind to [localhost]', 'localhost')
+    .option('--ssl', 'serve via https')
```
Comment (stefanpenner) [inline lib/cli.js]: Lets do:

```
  --ssl (Boolean) (Default: false)
  --ssl-key (String) (Default: ssl/server.key)
  --ssl-cert (String) (Default: ssl/server.crt)
```

File: lib/server.js [hunk 2]
```diff
@@ -9,21 +12,33 @@ function serve (builder, options) {
...
   server.watcher = options.watcher || new Watcher(builder)
 
   server.app = connect().use(middleware(server.watcher))
 
-  server.http = http.createServer(server.app)
+  if (options.ssl) {
+    var sslOptions
+    try {
+      sslOptions = {
+        key: fs.readFileSync(path.join(process.cwd(), 'ssl/server.key')),
```
Comment (stefanpenner) [inline lib/server.js]: `path.join(process.cwd()` here is not needed, `cwd` is implied when using `fs.*` methods

File: lib/server.js [hunk 3]
```diff
@@ -9,21 +12,33 @@ function serve (builder, options) {
...
+        cert: fs.readFileSync(path.join(process.cwd(), 'ssl/server.crt'))
+      }
+    } catch(e) {
+      throw new Error('SSL key and certificate files must be located at ssl/server.key and ssl/server.crt')
```
Comment (stefanpenner) [inline lib/server.js]: this error message likely should include `cwd` in the errors.

**Comment by stefanpenner** [comment]
@jakedetels sorry for the late review, if you can make the above changes. We can merge and :shipit:

**Comment by jakedetels** [comment]
@stefanpenner: I'll update this by end-of-month..

**Comment by rwjblue** [comment]
Friendly ping 😺

**Comment by stefanpenner** [comment]
landed via https://github.com/broccolijs/broccoli/pull/417

**Pull request closed without merging**